### DISCUSSION TITLE
Welcome to ol-mapbox-style Discussions!

### DISCUSSION THREAD

**Discussion by ahocevar** [openlayers/ol-mapbox-style#1046] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by orangemug** [comment]
I'm @orangemug, hello everyone 👋

I'm building a small open source mapping app for styling and manipulation of geo-data called _Maparatus_. The styling is much like Maputnik an uses the mapbox style specification as the basis for what possible. Unlike Maputnik which had openlayers as a "well it kind of works", my aim is to have openlayers as a fully operational renderer. It has a few bugs but basically works, and supports projections.

OpenLayers with `ol-mapbox-style` is a great library and I think for a bunch of mapping applications is a great fit. My aims for Maparatus are non-google-maps-esque maps, so not maps with "world level" to "street level" views, but maps with a lot more limits zoom limits. Those like you'd find on news websites for example. A little screenshot in case anybody is interested. It's a super-simple style in `EPSG:3031` projection.

<img src="https://github.com/openlayers/ol-mapbox-style/assets/235915/b9e66c87-cfda-4fc8-8b53-fec86d235f8a" width="400" />


Things I'd like to discuss (and maybe contribute)

 - A style specification in keeping with `maplibre-gl` that only exposes what's possible with `ol-mapbox-style` (also possibly extensions to the spec?)
 - Projection specific issues if they lie in openlayers or `ol-mapbox-style`
 - Feedback on a few "plugins" I've made for Maparatus that make them work together a little easier (making maplibre load fonts like ol-mapbox-style for example)
 - Any performance tips and tricks to make things run a bit smoother

Thanks everyone for all the hard work on the library, especially @ahocevar

**Comment by ahocevar** [comment]
Thanks for introducing yourself, @orangemug! What you're working on sounds great.

> A style specification in keeping with maplibre-gl that only exposes what's possible with ol-mapbox-style (also possibly extensions to the spec?)

I'm envisioning a standalone style spec package independent from Mapbox and MapLibre, where we can optimize things for build size on the one hand and auto-generated spec documentation with profiles and extensions on the other hand.

> Projection specific issues if they lie in openlayers or ol-mapbox-style

Instead of going down a path like WMTS or OGC API Tiles did, I'd llike to keep things as simple as the original TileJSON spec. This is what's already implemented in ol-mapbox-style. It means: validity extent of the used projection determines the tile grid, Web Mercator zoom levels determine the zoom levels for the styles.

**Reply by orangemug** [reply]
Thanks I've moved style-spec discussion into https://github.com/openlayers/ol-mapbox-style/issues/10#issuecomment-1850471621 

With regards to "projection specific issues", they're mostly working great thus far, here it is with a GeoJSON source from https://geojson.xyz/. Its come a long way since I last tried to implement this, congrats.

https://github.com/openlayers/ol-mapbox-style/assets/235915/2f39da4f-64c6-40e6-941d-23451ce01cd1

I've been using `EPSG:3031` for testing (which is probably one of the more awkward projections) I'm finding issues like with line widths, in the below screenshots, I'm increasing the line-width from `0.5` to `0.6` it gets interesting (artistic)

<img src="https://github.com/openlayers/ol-mapbox-style/assets/235915/4afbe563-3465-4ed2-bb49-4ab9d05f5ba1" width="300"/> <img src="https://github.com/openlayers/ol-mapbox-style/assets/235915/aed5a2ea-4260-4376-877b-16d84c3f507c" width="300" />
